gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/initializer/DefaultDashboardsInitializer.java

```diff
@@ -15,71 +15,68 @@
-            checkAndCreateDashboard(executionContext, API);
-            checkAndCreateDashboard(executionContext, APPLICATION);
-        }
-        return true;
-    }
 
-    private void checkAndCreateDashboard(ExecutionContext executionContext, final DashboardReferenceType referenceType) {
-        LOGGER.info("    No default {}'s dashboards, creating default ones...", referenceType);
-        createDashboard(executionContext, referenceType, "Global");
-        createDashboard(executionContext, referenceType, "Geo");
-        createDashboard(executionContext, referenceType, "User");
-        LOGGER.info("    Added default {}'s dashboards with success", referenceType);
+        findAllEnvironments()
+            .stream()
+            .filter(environment -> shouldConfigureDashboardForEnvironment(environment, dashboards))
+            .map(environment -> new ExecutionContext(environment.getOrganizationId(), environment.getId()))
+            .forEach(dashboardService::initialize);
+
+        return true;
     }
 
-    private void createDashboard(ExecutionContext executionContext, final DashboardReferenceType referenceType, final String prefixName) {
-        final NewDashboardEntity dashboard = new NewDashboardEntity();
-        dashboard.setName(prefixName + " dashboard");
-        dashboard.setReferenceId("DEFAULT");
-        dashboard.setReferenceType(io.gravitee.rest.api.model.DashboardReferenceType.valueOf(referenceType.name()));
-        dashboard.setEnabled(true);
-        final String filePath = format("/dashboards/%s_%s.json", referenceType.name().toLowerCase(), prefixName.toLowerCase());
+    private Set<EnvironmentEntity> findAllEnvironments() {
         try {
-            dashboard.setDefinition(IOUtils.toString(this.getClass().getResourceAsStream(filePath), defaultCharset()));
-        } catch (final Exception e) {
-            LOGGER.error("Error while trying to create a dashboard from the definition path: " + filePath, e);
+            return environmentService.findAllOrInitialize();
+        } catch (TechnicalManagementException e) {
+            log.warn("An error occurs searching all environments.");
         }
-        dashboardService.create(executionContext, dashboard);
+        return Set.of();
+    }
+
+    /**
+     * Checks if dashboards have been initialized for the environment
+     * @param environment is the reference to the environment to check if there are configured dashboards
+     * @param dashboards is the list of configured dashboards in database
+     * @return true if dashboards need to be initialized for the environment.
+     */
+    private static boolean shouldConfigureDashboardForEnvironment(EnvironmentEntity environment, List<DashboardEntity> dashboards) {
+        return dashboards
+            .stream()
+            .noneMatch(dashboard ->
+                dashboard.getReferenceType().equals("ENVIRONMENT") && dashboard.getReferenceId().equals(environment.getId())
```

Comment:
Can "ENVIRONMENT" be replaced by an enum value ?

Reply:
sure, the enum was not existing yet at this time, good catch